ADC_DMA/Core/Src/main.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[X]
+ * 	2 - Interrupt	[X]
+ * 	3 - DMA			[/]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+
+typedef struct {
+	uint16_t	v_ref;			//internal reference voltage measured (after calculation)
+	int16_t		temperature;	//temperature in Celsius degrees measured (after calculation)
+}sADC_measure;	//struct with all the information read by the ADC channel
+
+void 		wait(uint16_t time);	//delay function, system clock based
+
+int main(void)
+{
+	sADC_measure 	ADC_measure = {0};	//struct with all the information read by the ADC channel
+	uint16_t 		aux = 0;			//we only use aux to calculate v_ref
+
+	/*							  *
+	 * --- DMA INITIALIZATION --- *
+	 *							  */
+	RCC->AHBENR |= RCC_AHBENR_DMAEN;
+	DMA1_Channel1->CPAR	= 	(uint32_t) &(ADC1->DR);		//setting peripheral address
+	DMA1_Channel1->CCR 	|=	DMA_CCR_CIRC	|			//DMA in circular mode
+							DMA_CCR_MINC	|			//memory increment
+							DMA_CCR_PSIZE_0	|			//peripheral size set to 16 bits
+							DMA_CCR_MSIZE_0	;			//memory size set to 16 bits
+
+	/*										*
+	 * --- ADC INITIALIZATION PROCEDURE --- *
+	 *										*/
+	RCC->APB2ENR 	|= 	RCC_APB2ENR_ADC1EN;		//enables ADC clock
+	ADC1->CR 		|=	ADC_CR_ADVREGEN;		//enables the voltage regulator
+	ADC1->CFGR2 	|= 	ADC_CFGR2_CKMODE_0;		//sets ADC clock as PCLK/2
+	ADC->CCR 		|= 	ADC_CCR_LFMEN;			//low frequency mode enable
+	ADC1->SMPR 		|= 	ADC_SMPR_SMPR_2;		//sets sample time for 19.5 ADC clock cycles (14 needed)
```

Comment:
Just check if you are using the Sample time register right. Look if you don't have to specify the channel the you're setting the sample time.

Reply:
I don't remember to have noticed this before, so...

![Screen Shot 2020-06-01 at 13 26 01](https://user-images.githubusercontent.com/61191550/83430204-9d5dff00-a40b-11ea-8d4b-58a3276ab1d8.png)

Reply:
Ok! So it looks good to me!
Nice job!

Reply:
@gustavoscherer After your question is answered, you can click on "Resolve conversation" button :)

Reply:
@Vinimuller nice! I'm gonna to this now!